### PR TITLE
Review compositor tutorial

### PR DESCRIPTION
**Pull request by yhontyk**
Another docs review. I've provided a bit more context to the code creation part (at least as well as I myself could understand it).
Do you think the build definition of Cmake requires more explanations? Or will it be a self-evident thing for a regular C++ developer?

Jira: https://warthogs.atlassian.net/browse/DOCPR-852

### CODE REVIEW

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 1]
```diff
@@ -61,13 +64,17 @@ int main(int argc, char const* argv[])
...
-This program creates a floating window manager with basic window management
-capabilities such as controlling multiple windows, minimizing and maximizing,
-and handling mouse input. This is done with the help of MirAL (Mir Abstraction
-Layer) which gives you a high level interface to work with Mir.
 
-### Building
-To compile this simple program, you can use this `CMakeLists.txt` file:
+`MirRunner` is a class from the `libmiral` library that that provides a  high level interface for  Mir.
+
+`MirRunner runner` calls `run_with()` which specifies different fucntions of the composer. In this example, `run_with()` evokes a function `set_window_management_policy` that applies `MinimalWindowManager` policy to `runner`. The composer is therefore created with basic window management capabilities such as controlling multiple windows, minimizing and maximizing, and handling mouse input.
```
Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:70]:
```suggestion
`MirRunner runner` creates a `runner` object that can be used to configure your compositor.  To run the compositor you call `runner.run_with(...)` which specifying different functions of the composer. In this example, `run_with()` is passed a function `set_window_management_policy` that applies `MinimalWindowManager` policy to `runner`. The composer is therefore created with basic window management capabilities such as controlling multiple windows, minimizing and maximizing, and handling mouse input. 
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 2]
```diff
@@ -61,13 +64,17 @@ int main(int argc, char const* argv[])
...
-and handling mouse input. This is done with the help of MirAL (Mir Abstraction
-Layer) which gives you a high level interface to work with Mir.
 
-### Building
-To compile this simple program, you can use this `CMakeLists.txt` file:
+`MirRunner` is a class from the `libmiral` library that that provides a  high level interface for  Mir.
+
+`MirRunner runner` calls `run_with()` which specifies different fucntions of the composer. In this example, `run_with()` evokes a function `set_window_management_policy` that applies `MinimalWindowManager` policy to `runner`. The composer is therefore created with basic window management capabilities such as controlling multiple windows, minimizing and maximizing, and handling mouse input. 
+
+Through this syntax, you can add different functionality to the conposer such as different onscreen keyboards, screen capture, pointer confinement, and so on.
```
Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:72]:
```suggestion
Through the `runner`, you can add different functionality to the composer: enabling onscreen keyboards, screen capture, pointer confinement, and so on. 
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 3]
```diff
@@ -86,30 +93,29 @@ target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
...
+You can run a composer nested in an X or Wayland session, or from a virtual terminal, just like the demo applications in [Getting started with Mir](learn-what-mir-can-do.md). 
+
+For example, to run inside an existing Wayland session:
 ```sh
 WAYLAND_DISPLAY=wayland-99 ./build/demo-mir-compositor
 ```
-You'll see a window housing the compositor with a black void filling it. To
-fill this void with some content, you can run the following from another
-terminal:
+An all-black window with the compositor will pop up. Try moving it around the screen, maximizing and minimazing it. This functionality is provided by the `MinimalWindowManager` policy that you have added to your composer.
```
Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:109]:
```suggestion
An all-black window with the compositor will pop up.
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 4]
```diff
@@ -86,30 +93,29 @@ target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
...
-fill this void with some content, you can run the following from another
-terminal:
+An all-black window with the compositor will pop up. Try moving it around the screen, maximizing and minimazing it. This functionality is provided by the `MinimalWindowManager` policy that you have added to your composer. 
+
+To change the black background of the window and display some content instead, open another terminal and run:
 ```sh
 WAYLAND_DISPLAY=wayland-99 bomber
 ```
-You're free to replace `bomber` with any other Wayland compatible application.
+You can replace `bomber` with any other Wayland-compatible application.
```
Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:115]:
```suggestion
Try moving it around the screen, maximizing and minimazing it. This functionality is provided by the `MinimalWindowManager` policy that you have added to your composer. You can replace `bomber` with any other Wayland-compatible application.
```

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
> Do you think the build definition of Cmake requires more explanations? Or will it be a self-evident thing for a regular C++ developer?

CMake is the _de facto_ standard for C++ development. It is also fairly self explanatory: developers accustomed to different tooling will be able to extract the details they need.

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 5]
```diff
@@ -86,30 +93,29 @@ target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
...
-To run, you can run nested in an X or Wayland session, or from a virtual
-terminal, just like the demo applications in [Learn What Mir Can
-Do](learn-what-mir-can-do.md). For example, if we were to run inside an
-existing Wayland session, we'd run the following command:
+## Running a Mir composer
+You can run a composer nested in an X or Wayland session, or from a virtual terminal, just like the demo applications in [Getting started with Mir](learn-what-mir-can-do.md). 
+
+For example, to run inside an existing Wayland session:
 ```sh
 WAYLAND_DISPLAY=wayland-99 ./build/demo-mir-compositor
```
[the remark this answers is not in the document]
Comment (tarek-y-ismail) [reply doc/sphinx/tutorial/write-your-first-wayland-compositor.md:107]: Usually, Wayland compositors use `wayland-0` (or `wayland-1`, `wayland-2`, ... if you're running multiple compositors). This is used as the name of the compositor's socket which two compositors cannot share (as far as I know). So `wayland-99` is just a convention we use in the team to not have to worry about that.

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 6]
```diff
@@ -2,47 +2,50 @@
...
+
+This tutorial assumes that:
+
+* You are familiar with C++ and CMake.
+* You have `cmake` and a C++ compiler installed.
+
+## Installing dependencies
+
+The example program requires 
+* `libmiral`- a library for Mir abstraction layer
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:23]: Some suggestions: (looks like I misclicked something when starting the review :))

Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:23]:
```suggestion
* `libmiral` - short for "Mir Abstraction Layer". Provides a high level interface to interact with and customize Mir compositors.
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 7]
```diff
@@ -2,47 +2,50 @@
...
+* `mir-graphics-drivers-desktop` - a display server
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:24]:
```suggestion
* `mir-graphics-drivers-desktop` - provides drivers so Mir can talk with different graphics drivers
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 8]
```diff
@@ -2,47 +2,50 @@
...
 sudo apk add mir-dev
 ```
 </details>
 
 
-### Code
-The following code block is the bare minimum you need to run a Mir compositor:
+## Coding a Mir compositor
+
+The following code which defines a barebone Mir compositor:
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:47]:
```suggestion
The following code defines a barebones Mir compositor:
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 9]
```diff
@@ -61,13 +64,17 @@ int main(int argc, char const* argv[])
...
 ```
 
-This program creates a floating window manager with basic window management
-capabilities such as controlling multiple windows, minimizing and maximizing,
-and handling mouse input. This is done with the help of MirAL (Mir Abstraction
-Layer) which gives you a high level interface to work with Mir.
 
-### Building
-To compile this simple program, you can use this `CMakeLists.txt` file:
+`MirRunner` is a class from the `libmiral` library that that provides a  high level interface for  Mir.
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:68]:
```suggestion
`MirRunner` is a class from `libmiral` that acts as the "entry point" of your compositor.
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md (hunk 1 above)
Comment (tarek-y-ismail) [reply doc/sphinx/tutorial/write-your-first-wayland-compositor.md:70]: +1 on both comments, my version would be something like:

```suggestion
`MirRunner runner` creates a `runner` object that can be used to configure your compositor.  To run the compositor you call `runner.run_with(...)`, passing in different components to customize the behavior of the compositor. In this example, `run_with()` is passed a function `set_window_management_policy` that applies a `MinimalWindowManager` policy to the compositor. The compositor is therefore created with basic window management capabilities such as controlling multiple windows, minimizing and maximizing, and handling mouse input. 
```

**Review by tarek-y-ismail — CHANGES_REQUESTED** [review]

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 10]
```diff
@@ -2,47 +2,50 @@
...
 
-## Assumptions
-This tutorial assumes that:
-1. The reader is familiar with C++ and CMake.
-2. The reader has cmake and a C++ compiler installed
-
-## A barebones Mir compositor
 This section will cover the needed dependencies and how to install them, the
 minimum code needed for a Mir compositor, how to build this code, and finally
 how to run your compositor.
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:11]: I feel like these two paragraphs overlap a bit too much? I think only lines 5-7 are ncessary.

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 11]
```diff
@@ -2,47 +2,50 @@
...
+Install dependencies on Debian or derivatives:
 ```sh
 sudo apt install libmiral-dev mir-graphics-drivers-desktop
 ```
 
 <details>
 <summary> Installing Dependencies on Other Distros </summary>
 
-Installing dependencies on Fedora
+Instal dependencies on Fedora:
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:34]:
```suggestion
Install dependencies on Fedora:
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 12]
```diff
@@ -2,47 +2,50 @@
...
+* `mir-graphics-drivers-desktop` - provides drivers so Mir can talk with different graphics drivers
 
-Installing dependencies on Debian and its derivatives:
+Install dependencies on Debian or derivatives:
 ```sh
 sudo apt install libmiral-dev mir-graphics-drivers-desktop
 ```
 
 <details>
 <summary> Installing Dependencies on Other Distros </summary>
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:32]: This is inconsistent between this PR and #3605. I think hiding other distros might be better as to not overwhelm readers, but I'm open to other opinions.

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 13]
```diff
@@ -61,13 +64,17 @@ int main(int argc, char const* argv[])
...
 
-### Building
-To compile this simple program, you can use this `CMakeLists.txt` file:
+`MirRunner` is a class from `libmiral` that acts as the "entry point" of your compositor.
+
+`MirRunner runner` creates a `runner` object that can be used to configure your compositor.  To run the compositor you call `runner.run_with(...)`, passing in different components to customize the behavior of the compositor. In this example, `run_with()` is passed a function `set_window_management_policy` that applies a `MinimalWindowManager` policy to the compositor. The compositor is therefore created with basic window management capabilities such as controlling multiple windows, minimizing and maximizing, and handling mouse input. 
+
+The runner allows for even more customization: enabling onscreen keyboards, screen capture, pointer confinement, and so on. 
+
+## Building a Mir composer
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:74]:
```suggestion
## Building a Mir compositor
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 14]
```diff
@@ -86,30 +93,29 @@ target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
...
 cmake -B build
 cmake --build build
 ```
 
-### Running
-To run, you can run nested in an X or Wayland session, or from a virtual
-terminal, just like the demo applications in [Learn What Mir Can
-Do](learn-what-mir-can-do.md). For example, if we were to run inside an
-existing Wayland session, we'd run the following command:
+## Running a Mir composer
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:102]:
```suggestion
## Running a Mir compositor
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 15]
```diff
@@ -86,30 +93,29 @@ target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
...
+You can run a composer nested in an X or Wayland session, or from a virtual terminal, just like the demo applications in [Getting started with Mir](learn-what-mir-can-do.md).
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:103]:
```suggestion
You can run a compositor nested in an X or Wayland session, or from a virtual terminal, just like the demo applications in [Getting started with Mir](getting-started-with-mir.md). 
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 16]
```diff
@@ -86,30 +93,29 @@ target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
...
-fill this void with some content, you can run the following from another
-terminal:
+An all-black window with the compositor will pop up.
+
+To change the black background of the window and display some content instead, open another terminal and run:
 ```sh
 WAYLAND_DISPLAY=wayland-99 bomber
 ```
-You're free to replace `bomber` with any other Wayland compatible application.
+Try moving it around the screen, maximizing and minimazing it. This functionality is provided by the `MinimalWindowManager` policy that you have added to your composer. You can replace `bomber` with any other Wayland-compatible application.
```
Comment (tarek-y-ismail) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:115]:
```suggestion
Try moving it around the screen, maximizing and minimazing it. This functionality is provided by the `MinimalWindowManager` policy that you have added to your compositor. You can replace `bomber` with any other Wayland-compatible application.
```

**Review by tarek-y-ismail — CHANGES_REQUESTED** [review]

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 17]
```diff
@@ -2,47 +2,50 @@
...
-follows:
+## Assumptions
+
+This tutorial assumes that:
+
+* You are familiar with C++ and CMake.
+* You have `cmake` and a C++ compiler installed.
+
+## Installing dependencies
+
```
Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:21]: This section will cover the needed dependencies and how to install them.

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md (hunk 17 above)
Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:21]:
```suggestion
This section will cover the needed dependencies and how to install them.
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md (hunk 10 above)
Comment (AlanGriffiths) [reply doc/sphinx/tutorial/write-your-first-wayland-compositor.md:11]: Mostly agree. ("This section will cover the needed dependencies and how to install them." should move)

Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:11]:
```suggestion
```

File: doc/sphinx/tutorial/write-your-first-wayland-compositor.md [hunk 18]
```diff
@@ -86,30 +85,29 @@ target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
...
-fill this void with some content, you can run the following from another
-terminal:
+An all-black window with the compositor will pop up.
+
+To change the black background of the window and display some content instead, open another terminal and run:
 ```sh
 WAYLAND_DISPLAY=wayland-99 bomber
 ```
-You're free to replace `bomber` with any other Wayland compatible application.
+Try moving it around the screen, maximizing and minimazing it. This functionality is provided by the `MinimalWindowManager` policy that you have added to your compositor. You can replace `bomber` with any other Wayland-compatible application.
```
Comment (AlanGriffiths) [inline doc/sphinx/tutorial/write-your-first-wayland-compositor.md:107]:
```suggestion
Try moving it around the screen, maximizing and restoring it. This functionality is provided by the `MinimalWindowManager` policy that you have added to your compositor. You can replace `bomber` with any other Wayland-compatible application.
```

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Another typo

**Review by AlanGriffiths — APPROVED** [review]

**Review by tarek-y-ismail — APPROVED** [review]

**Comment by tarek-y-ismail** [comment]
@akcano & @yhontyk: Probably should've said this earlier, but thanks for your contributions!

**Pull request merged**